Core/Src/main.cpp

```diff
@@ -50,36 +32,56 @@ extern "C" void main_cpp(){
     mcuTemperatureTask.emplace();
     temperatureSensorsTask.emplace();
     timeKeepingTask.emplace();
-//    tcHandlingTask.emplace();
+    tcHandlingTask.emplace();
+    watchdogTask.emplace();
     canTestTask.emplace();
     canGatekeeperTask.emplace();
 
     uartGatekeeperTask->createTask();
     temperatureSensorsTask->createTask();
     mcuTemperatureTask->createTask();
     timeKeepingTask->createTask();
-//    tcHandlingTask->createTask();
+    tcHandlingTask->createTask();
+    watchdogTask->createTask();
     canTestTask->createTask();
     canGatekeeperTask->createTask();
+
     vTaskStartScheduler();
 
-    /**
-     * Uncomment below and comment above for Led task visualization (for STM32H743)
-     */
-//    xTaskCreate(blinkyTask1, "blinkyTask 2", 1000, nullptr, tskIDLE_PRIORITY + 1, nullptr);
-//    xTaskCreate(blinkyTask2, "blinkyTask 2", 1000, nullptr, tskIDLE_PRIORITY + 1, nullptr);
     for(;;);
     return;
 }
 
+/**
+ * HAL_UARTEx_RxEventCallback is a callback function for UART receiving which is enable by the
+ * the call of the function HAL_UARTEx_ReceiveToIdle_DMA().
+ * It is triggered whenever the UART line is IDLE.
+ * @param huart : pointer of type UART_HandleTypeDef
+ * @param Size : incoming size of data in bytes
+ */
+extern "C" void HAL_UARTEx_RxEventCallback(UART_HandleTypeDef *huart, uint16_t Size){
+    // Size is used for copying the right size of data to the TcCommand buffer,
+    // which belongs to the execute() of the tcHandlingTask
+    tcHandlingTask -> incomingMessageSize = Size;
+    BaseType_t xHigherPriorityTaskWoken;
+
+    // As always, xHigherPriorityTaskWoken is initialized to pdFALSE to be able to
+    // detect it getting set to pdTRUE inside an interrupt function
```

Comment:
```suggestion
```
This is standard FreeRTOS behavior so no further explaination is required from our side.